File: TASK-03/ruby/subtask4.rb
n = File.read("input.txt").to_i

File.open("output.txt", "w") do |outfile|
  (0..n/2).each do |i|
    outfile.puts(" " * (n / 2 - i) + "*" * (2 * i + 1))
  end
  (n/2 - 1).downto(0) do |i|
    outfile.puts(" " * (n / 2 - i) + "*" * (2 * i + 1))
  end
end

File: TASK-03/python/subtask4.py
with open("input.txt", "r") as infile:
    n = int(infile.read().strip())

with open("output.txt", "w") as outfile:
    for i in range(n // 2 + 1):
        outfile.write(" " * (n // 2 - i) + "*" * (2 * i + 1) + "\n")
    for i in range(n // 2 - 1, -1, -1):
        outfile.write(" " * (n // 2 - i) + "*" * (2 * i + 1) + "\n")
